govlearn-api/src/main/java/com/unimuenster/govlearnapi/course/repository/CourseRepository.java

```diff
@@ -47,28 +50,41 @@ public interface CourseRepository extends JpaRepository<Course, Long> {
     """)
     List<Course> findCoursesByCategory(int categoryId);
 
-    // @Query(value = """
-    //     SELECT c 
-    //     FROM Course c 
-    //     INNER JOIN c.courseTags t 
-    //     INNER JOIN t.tag tag
-    //     WHERE tag.id IN :tagIDs
-    //     AND ((lower(c.name) LIKE lower(concat('%', :nameSearch,'%'))) OR (lower(c.description) LIKE lower(concat('%', :nameSearch,'%'))))
-    // """)
+
     @Query(value = """
         SELECT DISTINCT (Course.*)
         FROM Course
         INNER JOIN course_tag ON course.id = course_tag.course_id
-        WHERE course_tag.tag_id IN :tagIDs
-        AND course.provider IN :Providers
-        AND course.skilllevel IN :Kompetenzstufe
-        AND course.format IN :Format
-        AND course.cost_free IN :Kosten
-        AND ((lower(course.name) LIKE lower(:nameSearch)) OR (lower(course.description) LIKE lower(:nameSearch)))
+        WHERE (:Providers is null or course.provider IN :Providers)
+        AND (:tagIDs is null or course_tag.tag_id IN :tagIDs or course_tag.tag_id is null)
```

Comment:
Es werden immer alle Kurse ausgegeben, deren Tags mit den gesuchten Tags mit min. einem Element übereinstimmen. Wollen wir genau die Tags haben, die wir in die Suche eingegeben haben (nicht mehr, nicht weniger), müssen wir nochmal tiefer rein.